hadoop-ozone/dist/src/main/smoketest/commonlib.robot

```diff
@@ -35,3 +41,51 @@ Compare files
     ${checksumbefore} =         Execute                    md5sum ${file1} | awk '{print $1}'
     ${checksumafter} =          Execute                    md5sum ${file2} | awk '{print $1}'
                                 Should Be Equal            ${checksumbefore}            ${checksumafter}
+Execute AWSS3APICli
+    [Arguments]       ${command}
+    ${output} =       Execute                    aws s3api --endpoint-url ${ENDPOINT_URL} ${command}
+    [return]          ${output}
+
+Execute AWSS3APICli and checkrc
+    [Arguments]       ${command}                 ${expected_error_code}
+    ${output} =       Execute and checkrc        aws s3api --endpoint-url ${ENDPOINT_URL} ${command}  ${expected_error_code}
+    [return]          ${output}
+
+Execute AWSS3Cli
+    [Arguments]       ${command}
+    ${output} =       Execute                     aws s3 --endpoint-url ${ENDPOINT_URL} ${command}
+    [return]          ${output}
+
+Install aws cli s3 centos
+    Execute            sudo yum install -y awscli
+
+Install aws cli s3 debian
+    Execute            sudo apt-get install -y awscli
+
+Install aws cli
+    ${rc}              ${output} =                 Run And Return Rc And Output           which apt-get
+    Run Keyword if     '${rc}' == '0'              Install aws cli s3 debian
+    ${rc}              ${output} =                 Run And Return Rc And Output           yum --help
+    Run Keyword if     '${rc}' == '0'              Install aws cli s3 centos
+
+Kinit test user
+    ${hostname} =       Execute                    hostname
+    Set Suite Variable  ${TEST_USER}               testuser/${hostname}@EXAMPLE.COM
+    Execute             kinit -k ${TEST_USER} -t /etc/security/keytabs/testuser.keytab
+
+Setup secure credentials
+    Run Keyword         Install aws cli
```

Comment:
Do we need to install aws cli for all the tests? I think most of the time "Kinit test user" is enough

Reply:
currently it used to setup aws secure credentials.

Reply:
you are right, moved s3 part to s3 commonlib.